src/private_broadcast.cpp

```diff
@@ -14,10 +15,79 @@ void PrivateBroadcast::Add(const CTransactionRef& tx) EXCLUSIVE_LOCKS_REQUIRED(!
     }
 }
 
+std::optional<CTransactionRef> PrivateBroadcast::GetTxForBroadcast() EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    if (m_by_priority.empty()) {
+        return std::nullopt;
+    }
+    const Txid& txid = m_by_priority.begin()->second;
+    auto it = m_by_txid.find(txid);
+    if (Assume(it != m_by_txid.end())) {
+        return it->second.tx;
+    }
+    m_by_priority.erase(m_by_priority.begin());
```

Comment:
b81c3cc19ec8055d04b38a72fdfabbfcb2ed2358

Is it ok to erase from this map now? What if none of our connections are successful ?

Reply:
Note that this would only be reached if `m_by_priority` contains txid that does not have an entry in `m_by_txid` which is a gross bug in the `PrivateBroadast` class. IIRC this was an assert before but somebody suggested to use `Assume()` instead.